csrank/tests/test_choice_functions.py

```diff
@@ -1,98 +1,87 @@
 import pytest
-import tensorflow as tf
+import torch
+from torch import optim
 
-from csrank.choicefunction import *
-from csrank.constants import CMPNET_CHOICE
+from csrank.choicefunction import FATEChoiceFunction
+from csrank.choicefunction import GeneralizedLinearModel
+from csrank.choicefunction import PairwiseSVMChoiceFunction
 from csrank.constants import FATE_CHOICE
-from csrank.constants import FATELINEAR_CHOICE
-from csrank.constants import FETA_CHOICE
-from csrank.constants import FETALINEAR_CHOICE
 from csrank.constants import GLM_CHOICE
-from csrank.constants import RANKNET_CHOICE
 from csrank.constants import RANKSVM_CHOICE
 from csrank.metrics_np import auc_score
 from csrank.metrics_np import f1_measure
 from csrank.metrics_np import instance_informedness
 from csrank.metrics_np import subset_01_loss
+from csrank.modules.zeroth_order import DenseNeuralNetwork
 from csrank.util import metrics_on_predictions
 
 choice_metrics = {
     "F1Score": f1_measure,
     "Informedness": instance_informedness,
     "AucScore": auc_score,
 }
-optimizer_common_args = {
-    "optimizer": SGD,
-    "optimizer__lr": 1e-3,
-    "optimizer__momentum": 0.9,
-    "optimizer__nesterov": True,
-}
 
 
 def get_vals(values):
     return dict(zip(choice_metrics.keys(), values))
 
 
 choice_functions = {
-    FETA_CHOICE: (
-        FETAChoiceFunction,
-        {"add_zeroth_order_model": True, **optimizer_common_args},
-        get_vals([0.946, 0.9684, 0.9998]),
-    ),
+    GLM_CHOICE: (GeneralizedLinearModel, {}, get_vals([0.9567, 0.9955, 1.0])),
+    RANKSVM_CHOICE: (PairwiseSVMChoiceFunction, {}, get_vals([0.9522, 0.9955, 1.0])),
     FATE_CHOICE: (
         FATEChoiceFunction,
         {
-            "n_hidden_joint_layers": 1,
-            "n_hidden_set_layers": 1,
-            "n_hidden_joint_units": 5,
-            "n_hidden_set_units": 5,
-            **optimizer_common_args,
+            "max_epochs": 100,
+            "optimizer": optim.SGD,
+            "optimizer__lr": 1e-3,
+            "optimizer__momentum": 0.9,
+            "optimizer__nesterov": True,
+            "module__pairwise_utility_module": functools.partial(
+                DenseNeuralNetwork,
+                hidden_layers=1,  # n_hidden_joint_layers
+                units_per_hidden=5,  # n_hidden_joint_units
+                output_size=1,
+            ),
+            "module__zeroth_order_embedding_module": functools.partial(
+                # n_hidden_set_layers, n_hidden_set_units
+                DenseNeuralNetwork,
+                hidden_layers=1,
+                units_per_hidden=5,
+            ),
         },
-        get_vals([0.8185, 0.6070, 0.9924]),
-    ),
-    FATELINEAR_CHOICE: (
-        FATELinearChoiceFunction,
-        {},
-        get_vals([0.6558, 0.0722, 0.9998]),
-    ),
-    FETALINEAR_CHOICE: (
-        FETALinearChoiceFunction,
-        {},
-        get_vals([0.8782, 0.8894, 0.9998]),
+        get_vals([0.6861, 0.2014, 1.0]),
```

Comment:
The F1 score and the instance informedness went down quite a bit for the FATE general choice model. If I increase the network size slightly (10 instead of 5 for `n_hidden_joint_layers`) the performance goes up again though (`[0.9019, 0.91043, 1.0]`). Most of it is probably just random variation since the example is very small. I think the test is mostly for the basic functionality and the exact numbers do not matter very much here, but I still wanted to highlight the change and start a discussion.